app/lib/govuk_elements_form_builder/form_builder.rb

```diff
@@ -0,0 +1,89 @@
+module GovukElementsFormBuilder
+  class FormBuilder < ActionView::Helpers::FormBuilder
+    # Prevents surrounding of erroneous inputs with <div class="field_with_errors">
+    # https://guides.rubyonrails.org/configuring.html#configuring-action-view
+    ActionView::Base.field_error_proc = proc { |html_tag| html_tag.html_safe }
+
+    delegate :content_tag, to: :@template
+    delegate :errors, to: :@object
+
+    def govuk_text_field(attribute, *args)
+      content_tag :div, class: form_group_classes(attribute) do
+        options = args.extract_options!
+
+        label_classes!(options)
+        field_classes!(options, attribute)
+
+        form_group(options, attribute)
+      end
+    end
+
+    private
+
+    def form_group(options, attribute)
+      html = label(attribute, options[:label_options].merge(for: attribute))
+
+      html.concat(hint_tag(attribute)) if !options[:hide_hint?] && hint_message(attribute)
+      html.concat(error_tag(attribute)) if error?(attribute)
+
+      (html + text_field(attribute, options.except(:label, :label_options).merge(id: attribute))).html_safe
+    end
+
+    # Given an attributes hash that could include any number of arbitrary keys, this method
+    # ensure we merge one or more 'default' attributes into the hash, creating the keys if
+    # don't exist, or merging the defaults if the keys already exists.
+    # It supports strings or arrays as values.
+    #
+    def merge_attributes(attributes, default:)
+      hash = attributes || {}
+      hash.merge(default) { |_key, oldval, newval| Array(newval) + Array(oldval) }
+    end
+
+    def field_classes!(options, attribute)
+      default_classes = ['govuk-input']
+      default_classes << 'govuk-input--error' if error?(options[:field_with_error] || attribute)
+
+      options ||= {}
+      options.merge!(
+        merge_attributes(options, default: { class: default_classes })
+      )
+    end
+
+    def label_classes!(options)
+      options ||= {}
+      options[:label_options] ||= {}
+      options[:label_options].merge!(
+        merge_attributes(options[:label_options], default: { class: 'govuk-label' })
+      )
```

Comment:
I think if you use `merge!` within `merge_attributes` this can be replaced with just:
```ruby
merge_attributes(options[:label_options], default: { class: 'govuk-label' })
```
No outer `options[:label_options].merge!`